### ISSUE TITLE
mermaid DAG

### ISSUE THREAD

**Issue by wlandau** [ropensci/targets#775]
## Prework

* [x] Read and agree to the [code of conduct](https://ropensci.org/code-of-conduct/) and [contributing guidelines](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/ropensci/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.
* [x] Format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Proposal

https://twitter.com/yoniceedee/status/1493635555355152392, @yonicd

**Comment by yonicd** [comment]
cool. thanks!

**Comment by wlandau** [comment]
Simple enough syntax:

```mermaid
graph LR
  classDef built color:black,fill:#E1BD6D;
  classDef uptodate stroke-width:0px,color:white,fill:#354823;
  classDef outdated stroke-width:0px,color:black,fill:#78B7C5;
  classDef started stroke-width:0px,color:black,fill:#DC863B;
  classDef canceled stroke-width:0px,color:black,fill:#FAD510;
  classDef errored stroke-width:0px,color:white,fill:#C93312;
  classDef queued stroke-width:0px,color:black,fill:#D2D2D0;
  classDef skipped stroke-width:0px,color:white,fill:#7500D1;
  classDef none stroke-width:0px,color:black,fill:#94a4ac;
  o[\object/]:::uptodate --> a([static]):::uptodate
  f[/function\]:::uptodate --> a([static]):::outdated
  a([static]):::uptodate --> b>dynamic]:::errored
```

**Comment by yonicd** [comment]
🤩

**Comment by wlandau** [comment]
Here's an auto-generated one from the code I am working on. I love that I don't have to micromanage the positions and it just looks good.

```mermaid
graph LR
  classDef outdated stroke-width:0px,color:#000000,fill:#78B7C5;
  classDef uptodate stroke-width:0px,color:#ffffff,fill:#354823;
  g>g]:::uptodate --> f>f]:::uptodate
  data1([data1]):::outdated --> map1[map1]:::outdated
  data1([data1]):::outdated --> map2[map2]:::outdated
  data1([data1]):::outdated --> out1([out1]):::outdated
  data1([data1]):::outdated --> out2([out2]):::outdated
  out6([out6]):::outdated --> out3([out3]):::outdated
  cross1[cross1]:::outdated --> out4([out4]):::outdated
  cross2[cross2]:::outdated --> out4([out4]):::outdated
  cross3[cross3]:::outdated --> out5([out5]):::outdated
  cross3[cross3]:::outdated --> out5([out5]):::outdated
  out1([out1]):::outdated --> out6([out6]):::outdated
  out2([out2]):::outdated --> cross1[cross1]:::outdated
  out2([out2]):::outdated --> cross2[cross2]:::outdated
  out3([out3]):::outdated --> cross2[cross2]:::outdated
  data2([data2]):::outdated --> cross3[cross3]:::outdated
  map1[map1]:::outdated --> cross3[cross3]:::outdated
```

**Comment by yonicd** [comment]
That is really nice. My goal is to leverage this to create an analysis readme page so collaborators can go into a pipeline subdir in a gh repo and see the pipeline before starting to work with it. I am also thinking how best to create a targets bookdown type output that each pipeline is a chapter and it is used as documentation for a targets based repo.
